File: fortran/objshear/test/test-config.f90
! vim:set ft=fortran:
program main

    use configlib

    character*256 filename
    type(config) pars


    if (iargc() /= 1) then
        print *,"Usage: test-config filename"
        stop 45
    end if


    call getarg(1,filename)

    call read_config(filename, pars)
    call print_config(pars)

end program

File: ccode/objshear2/test/test-config.cc
#include <stdio.h>
#include <stdlib.h>
#include "../config.h"


int main(int argc, char** argv) {

    if (argc != 2 ) {
        printf("usage: test-config filename\n");
        exit(45);
    }

    const char* filename = argv[1];

    struct config pars;

    read_config(filename, pars);
    print_config(pars);
}

